### PR TITLE
Changed Path AST node to use its root node source location

### PR DESCRIPTION
**Pull request by lziq**
*Issue #526*

*Description of changes:*
Changed Path AST node to use its root node source location

Now the error becomes what we expected: 
Screen Shot 2022-02-11 at 12 09 08 PM

### CODE REVIEW

File: lang/test/org/partiql/lang/eval/visitors/StaticTypeInferenceVisitorTransformTest.kt [hunk 1]
```diff
@@ -5737,6 +5737,21 @@ class StaticTypeInferenceVisitorTransformTest : VisitorTransformTestBase() {
...
+                    expectedProblems = listOf(
+                        createInvalidArgumentTypeForFunctionError(
+                            sourceLocation = SourceLocationMeta(1L, 7L, 1L),
+                            functionName = "upper",
+                            expectedArgType = unionOf(STRING, SYMBOL),
+                            actualType = INT
+                        )
+                    )
+                )
+            ),
```
Comment (alancai98) [inline lang/test/org/partiql/lang/eval/visitors/StaticTypeInferenceVisitorTransformTest.kt:5754]: Could you also add the similar path test case brought up by the customer involving list references of the wrong type:

Something like `UPPER(x[0])` with a globals of `"x" to ListType(INT)`?

**Review by alancai98 — CHANGES_REQUESTED** [review]
Fix looks good to me! Have one comment to add the other path test case brought up by the customer.

File: lang/test/org/partiql/lang/eval/visitors/StaticTypeInferenceVisitorTransformTest.kt (hunk 1 above)
Comment (lziq) [reply lang/test/org/partiql/lang/eval/visitors/StaticTypeInferenceVisitorTransformTest.kt:5754]: Will add the test case in the next commit.

**Review by alancai98 — APPROVED** [review]
:shipit:

**Pull request merged**